File: internal/model/login_provider.go
package model

import "strings"

const (
	LoginGoogle    = "google"
	LoginMicrosoft = "microsoft"
)

func NormalizeLoginProvider(raw string) string {
	switch strings.ToLower(strings.TrimSpace(raw)) {
	case LoginMicrosoft, "ms", "outlook", "hotmail", "live":
		return LoginMicrosoft
	default:
		return LoginGoogle
	}
}

func (a Account) IsMicrosoft() bool {
	return NormalizeLoginProvider(a.LoginProvider) == LoginMicrosoft
}

File: web/src/lib/login-provider.ts
export type LoginProvider = "google" | "microsoft"

export function normalizeLoginProvider(raw?: string): LoginProvider {
  const value = (raw || "").trim().toLowerCase()
  if (value === "microsoft" || value === "ms" || value === "outlook" || value === "hotmail" || value === "live") {
    return "microsoft"
  }
  return "google"
}

export function loginProviderLabel(raw?: string) {
  return normalizeLoginProvider(raw) === "microsoft" ? "微软" : "谷歌"
}
